core/src/main/kotlin/org/neo4j/graphql/handler/projection/ProjectionBase.kt

```diff
@@ -215,7 +215,7 @@ open class ProjectionBase {
         val cypherDirective = fieldDefinition.cypherDirective()
         val isObjectField = fieldDefinition.type.inner() is GraphQLFieldsContainer
         return cypherDirective?.let {
-            val directive = cypherDirective(variable, fieldDefinition, field, it, listOf(Translator.CypherArgument("this", "this", variable)))
+            val directive = cypherDirective(variable + field.aliasOrName().capitalize(), fieldDefinition, field, it, listOf(Translator.CypherArgument("this", "this", variable)))
```

Comment:
do we have the same problem in other places? 
sounds like a good idea to encapsulate this variable-contextualization into a method on field or so:

field.contextualize(variable) or field.for(variable) ? perhaps you can do that in a separate PR